snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedRefSetSearchRequestBuilder.java

```diff
@@ -82,6 +82,17 @@ public SnomedRefSetSearchRequestBuilder filterByMapTargetComponentType(Integer m
 		}
 		return addOption(SnomedRefSetSearchRequest.OptionKey.MAP_TARGET_COMPONENT_TYPE, mapTargetComponentType);
 	}
+	
+	public SnomedRefSetSearchRequestBuilder filterByMapTargetComponentType(String mapTargetComponentType) {
+		if (Strings.isNullOrEmpty(mapTargetComponentType)) {
```

Comment:
Never ever put conditional blocks into `filterBy` methods.
Handle all these in the actual `Request` implementation instead.
Not sure if these are actually needed.
This way we cannot filter reference sets where the `mapTargetComponentType` is `UNSPECIFIED`.
Null values are the only values that do NOT generate actual clauses in search requests.
Please refactor.

Reply:
Done